source/references.html

```diff
@@ -351,19 +379,90 @@ <h3>Canonical URLs</h3>
- <li></li>
+ <li>If there are multiple matches (because the version is missing, or incomplete - see below) then it is a matter of policy to decide how to resolve this (see below)</li>
 </ul>
 <p>
-Systems resolving references to canonical URLs SHOULD first try to resolve the reference using
-the canonical reference (e.g. search on a known registry of terminology, conformance, or knowledge
-resources as appropriate), and then fall back to direct resolution using the URL as
-a literal reference if a local version of the canonical resource cannot be found.
-This approach is safe because the approaches must refer to the same
-artifact, though implementations will need to make appropriate arrangements regarding
-the version and/or currency of their local copy of the artifact.
+Systems resolving references to canonical URLs SHOULD first try to , and then .
+
 </p>
+
+<a name="matching"></a>
+<h3>Version Matching for Canonical Resources</h3>
+<p>
+In order to work with Canonical Resources, it is necessary to be able to match version strings
+from versioned references to canonical resources to the resources themselves. Among other places, 
+versioned references arise in the following contexts:
+</p>
+<ul>
+ <li>Any Canonical Reference</li>
+ <li>Coding.system+version</li>
+ <li>ValueSet.compose.include.system+version</li>
+ <li>Dependencies between <a href="packages.html">packages</a></li>
+</ul>
+<p>
+If there is no stated version for the reference, or the version is a wildcard (see below), 
+<b>and if the reference is found in a resource in an <a href="packages.html">NPM Package</a></b>, 
+then any application looking for the correct version SHALL first look through the package and its dependencies 
+when resolving the reference. If no matching version is found in the package or its dependencies, the application
+can then look through any other resources it has available. This rule applies even once the resource is removed 
+from the package. Other ways target resources might be made available to applications:
+</p>
+<ul>
+  <li>The resource might be made available to the system through it's normal configuration</li>
+  <li>The application might resolve the reference using the canonical reference (e.g. search on a known registry
+    of terminology, conformance, or knowledge resources as appropriate)</li>
+  <li>The application might fall back to direct resolution using the URL as a literal reference if a local
+    version of the canonical resource cannot be found. Note: this approach is safe because the approaches
+    must refer to the same artifact, though implementations will need to make appropriate arrangements regarding
+    the version and/or currency of their local copy of the artifact.</li>
+</ul>
+<p>
+Applications are not required to look through external references, and production systems typically do not.

```

Comment:
It's not clear (to me at least) what is meant by "external references" here?

Reply:
duh. should've been external sources